### DISCUSSION TITLE
adding custom id/attributes to sprite/elements

### DISCUSSION THREAD

**Discussion by k11q** [pixijs/pixijs#9000] [Q&A] [answered]
Hi I am figuring out how to add custom id to my sprite/elements so I can identify when clicking them to change my data. Is there any resources on this?

Also I am using vue, how would I bind a dynamic data to each element, so when I change data the width/height also change for example.

Thank you.

**Comment by miltoncandelero** [comment, marked as answer]
all display objects have a `name` field where you can put any string. (if you twist javascript arm, you can put actually anything inside it)

otherwise you will need to extend and create your own mixin 🤔 
try and see how `name` is added via the mixin

inside a typescript `.d.ts` file
```
declare namespace GlobalMixins
{
    interface DisplayObject
    {
        whateverYouWantHere: any
    }
}
```

(if you are not using typescript, just jam something inside the object, javascript wont complain 🤣 )

**Reply by k11q** [reply]
you are a champion. Thank you so much!

**Reply by k11q** [reply]
sorry to bother, another question, how to find my pixi app instance? since I create it in another function and don't use global variable, I want to access it, how may I do it? I dont want to use global variable

**Reply by miltoncandelero** [reply]
The only way to have global access to your `app` instance is to have a global variable.
You can hide it inside a static class, a singleton class, a service locator, you name it. But pixi doesn't track your created `Application` instance
